sw/device/lib/drivers/rv_plic/rv_plic.c

```diff
@@ -34,337 +81,417 @@ typedef struct plic_reg_info {
  * accommodate all the bits (1 bit per IRQ source). This function calculates
  * the offset for a specific IRQ source ID (ID 32 would be IE01, ...).
  */
-static ptrdiff_t plic_offset_from_reg0(dif_plic_irq_id_t irq) {
-  uint8_t register_index = irq / RV_PLIC_PARAM_REG_WIDTH;
-  return register_index * sizeof(uint32_t);
-}
+static ptrdiff_t plic_offset_from_reg0(plic_irq_id_t irq);
 
 /**
+ * 
  * Get an IE, IP, LE register bit index from an IRQ source ID.
  *
  * With more than 32 IRQ sources, there is a multiple of these registers to
  * accommodate all the bits (1 bit per IRQ source). This function calculates
  * the bit position within a register for a specific IRQ source ID (ID 32 would
  * be bit 0).
  */
-static uint8_t plic_irq_bit_index(dif_plic_irq_id_t irq) {
-  return irq % RV_PLIC_PARAM_REG_WIDTH;
+static uint8_t plic_irq_bit_index(plic_irq_id_t irq);
+/****************************************************************************/
+/**                                                                        **/
+/*                           EXPORTED VARIABLES                             */
+/**                                                                        **/
+/****************************************************************************/
+
+/****************************************************************************/
+/**                                                                        **/
+/*                            GLOBAL VARIABLES                              */
+/**                                                                        **/
+/****************************************************************************/
+
+int8_t external_intr_flag = 0;
```

Comment:
why is it signed? can you put a comment to specify the context of this variable (what is its purpose etc?)?